File: 008/Python/Problem008Tests.py
#!/usr/bin/env python

import unittest

import problem008

class Problem008Tests(unittest.TestCase):

    def testlargest_five_digit_product(self):
        self.assertEquals(0, problem008.largest_five_digit_product(1234))
        self.assertEquals(1890, problem008.largest_five_digit_product(23579))
        self.assertEquals(1890, problem008.largest_five_digit_product(123579))
        self.assertEquals(1890, problem008.largest_five_digit_product(235791))
        self.assertEquals(3125, problem008.largest_five_digit_product(223355555337))
        self.assertEquals(1890, problem008.largest_five_digit_product(123223957112110))
        self.assertEquals(32, problem008.largest_five_digit_product(1111222221111113))

unittest.main()

File: 008/Erlang/problem008_tests.erl
-module(problem008_tests).

-include_lib("eunit/include/eunit.hrl").
-import(problem008).

largest_five_digit_product_test_() -> [
  ?_assertEqual(0, problem008:largest_five_digit_product(1234)),
  ?_assertEqual(1890, problem008:largest_five_digit_product(23579)),
  ?_assertEqual(1890, problem008:largest_five_digit_product(123579)),
  ?_assertEqual(1890, problem008:largest_five_digit_product(235791)),
  ?_assertEqual(3125, problem008:largest_five_digit_product(223355555337)),
  ?_assertEqual(1890, problem008:largest_five_digit_product(123223957112110)),
  ?_assertEqual(32, problem008:largest_five_digit_product(1111222221111113)),
  ?_assert(true)
  ].

